File: genius-bi-ui/src/services/analysis-assistants.ts
import api from './api';
import type { AnalysisAssistant, ApiResponse, PaginatedResponse } from '../types';

export const analysisAssistantsService = {
  // Get all analysis assistants with pagination
  getAnalysisAssistants: async (page: number = 1, size: number = 10) => {
    const response = await api.get<PaginatedResponse<AnalysisAssistant>>('/analysis-assistants', {
      params: { page, size }
    });
    return response.data;
  },

  // Get a single analysis assistant by ID
  getAnalysisAssistant: async (id: number) => {
    const response = await api.get<ApiResponse<AnalysisAssistant>>(`/analysis-assistants/${id}`);
    return response.data;
  },

  // Create a new analysis assistant
  createAnalysisAssistant: async (assistant: Omit<AnalysisAssistant, 'id' | 'created_at' | 'updated_at'>) => {
    const response = await api.post<ApiResponse<AnalysisAssistant>>('/analysis-assistants', assistant);
    return response.data;
  },

  // Update an existing analysis assistant
  updateAnalysisAssistant: async (id: number, assistant: Partial<AnalysisAssistant>) => {
    const response = await api.put<ApiResponse<AnalysisAssistant>>(`/analysis-assistants/${id}`, assistant);
    return response.data;
  },

  // Delete an analysis assistant
  deleteAnalysisAssistant: async (id: number) => {
    const response = await api.delete<ApiResponse<void>>(`/analysis-assistants/${id}`);
    return response.data;
  }
}; 

File: genius-bi-server/src/routers/analysis_assistants.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List

from models import AnalysisAssistant as AnalysisAssistantModel
from schemas import AnalysisAssistant as AnalysisAssistantSchema
from schemas import AnalysisAssistantCreate, AnalysisAssistantUpdate
from database import get_db

router = APIRouter()

@router.post("/analysis-assistants/")
def create_analysis_assistant(assistant: AnalysisAssistantCreate, db: Session = Depends(get_db)):
    db_assistant = AnalysisAssistantModel(**assistant.model_dump())
    db.add(db_assistant)
    db.commit()
    db.refresh(db_assistant)
    return {"data": db_assistant}

@router.get("/analysis-assistants/")
def read_analysis_assistants(page: int = 0, limit: int = 100, db: Session = Depends(get_db)):
    # Calculate skip based on page number and limit
    skip = (page - 1) * limit

    # Get the total number of terms
    total_terms = db.query(AnalysisAssistantModel).count()

    # Get the terms for the current page
    terms = db.query(AnalysisAssistantModel).offset(skip).limit(limit).all()

    # Calculate total pages
    total_pages = (total_terms + limit - 1) // limit if total_terms > 0 else 1

    return {
        "items": terms,
        "pages": total_pages,
        "page": page
    }

@router.get("/analysis-assistants/{assistant_id}")
def read_analysis_assistant(assistant_id: int, db: Session = Depends(get_db)):
    assistant = db.query(AnalysisAssistantModel).filter(AnalysisAssistantModel.id == assistant_id).first()
    if assistant is None:
        raise HTTPException(status_code=404, detail="Analysis assistant not found")
    return {"data": assistant}

@router.put("/analysis-assistants/{assistant_id}", response_model=AnalysisAssistantSchema)
def update_analysis_assistant(assistant_id: int, assistant: AnalysisAssistantUpdate, db: Session = Depends(get_db)):
    db_assistant = db.query(AnalysisAssistantModel).filter(AnalysisAssistantModel.id == assistant_id).first()
    if db_assistant is None:
        raise HTTPException(status_code=404, detail="Analysis assistant not found")
    for var, value in vars(assistant).items():
        setattr(db_assistant, var, value)
    db.commit()
    db.refresh(db_assistant)
    return db_assistant

@router.delete("/analysis-assistants/{assistant_id}", response_model=AnalysisAssistantSchema)
def delete_analysis_assistant(assistant_id: int, db: Session = Depends(get_db)):
    db_assistant = db.query(AnalysisAssistantModel).filter(AnalysisAssistantModel.id == assistant_id).first()
    if db_assistant is None:
        raise HTTPException(status_code=404, detail="Analysis assistant not found")
    db.delete(db_assistant)
    db.commit()
    return db_assistant 